elliottlib/cli/change_state_cli.py

```diff
@@ -56,53 +59,53 @@ def change_state_cli(runtime, state, advisory, default_advisory_type, noop):
-            if default_advisory_type is not None:
-                raise ElliottFatalError("Error: Could not change '{state}' advisory {advs}, group.yml is probably pointing at old one".format(state=e.errata_state, advs=advisory))
+    if advisory:
+        advisories.append(advisory)
+
+    if default_advisories:
+        for impetus in ["rpm", "image", "extras"]:
+            advisories.append(runtime.group_config.advisories[impetus])
+
+    for advisory in advisories:
+        try:
+            e = Erratum(errata_id=advisory)
+
+            if e.errata_state == state:
+                green_prefix(f"No Change: {advisory}: ")
+                click.echo(f"Target state is same as current state: {state}")
+            # we have 5 different states we can only change the state if it's in NEW_FILES or QE
+            # "NEW_FILES",
+            # "QE",
+            # "REL_PREP",
+            # "PUSH_READY",
+            # "IN_PUSH"
+            elif e.errata_state != 'NEW_FILES' and e.errata_state != 'QE':
+                red_prefix(f"Error: {advisory}: ")
+                if default_advisory_type is not None:
+                    click.echo(f"Could not change '{e.errata_state}', group.yml is probably pointing at old one")
+                else:
+                    click.echo(f"Can only change the state if it's in NEW_FILES or QE, current state is {e.errata_state}")
             else:
-                raise ElliottFatalError("Error: we can only change the state if it's in NEW_FILES or QE, current state is {s}".format(s=e.errata_state))
-        else:
-            if noop:
-                green_prefix("{}Would have changed state: ".format(prefix))
-                click.echo("{} ➔ {}".format(e.errata_state, state))
-                return
-            else:
-                # Capture current state because `e.commit()` will
-                # refresh the `e.errata_state` attribute
-                old_state = e.errata_state
-                e.setState(state)
-                e.commit()
-                green_prefix("Changed state: ")
-                click.echo("{old_state} ➔ {new_state}".format(
-                    old_state=old_state,
-                    new_state=state))
-    except ErrataException as ex:
-        raise ElliottFatalError(getattr(ex, 'message', repr(ex)))
-
-    green_print("Successfully changed advisory state")
+                if noop:
+                    green_prefix(f"NOOP: {advisory}: ")
+                    click.echo(f"Would have changed state {e.errata_state} ➔ {state}")
+                else:
+                    # Capture current state because `e.commit()` will
+                    # refresh the `e.errata_state` attribute
+                    old_state = e.errata_state
+                    e.setState(state)
+                    e.commit()
+                    green_prefix("Changed state: ")
+                    click.echo(f"{old_state} ➔ {state}")
+        except ErrataException as ex:
+            print(ex)
```

Comment:
This will mask the error to the caller. Please re-raise the exception so that the caller of Elliott will notice if there's an error.

Reply:
Thanks, I collected all the errors and raising them at the end